---
# Cannot use Block/Rescue in Ansible Handlers: https://github.com/ansible/ansible/issues/14270
# Implementing try/catch logic with ignore_errors and conditionals
- name: Wait for API to return 200
  uri:
    url: "{{ksql_http_protocol}}://{{inventory_hostname}}:{{ksql_listener_port}}/info"
    status_code: 200
    validate_certs: false
    # Basic auth header is redundant for non rbac install
    url_username: "{{ksql_ldap_user}}"
    url_password: "{{ksql_ldap_password}}"
    force_basic_auth: true
  register: result1
  until: result1.status == 200
  retries: 40
  delay: 5
  when: not ksql_ssl_mutual_auth_enabled|bool
  ignore_errors: true

- name: Wait for API to return 200 - mTLS
  uri:
    url: "{{ksql_http_protocol}}://{{inventory_hostname}}:{{ksql_listener_port}}/info"
    status_code: 200
    validate_certs: false
    client_cert: "{{ksql_cert_path}}"
    client_key: "{{ksql_key_path}}"
    # Basic auth header is redundant for non rbac install
    url_username: "{{ksql_ldap_user}}"
    url_password: "{{ksql_ldap_password}}"
    force_basic_auth: true
  register: result2
  until: result2.status == 200
  retries: 40
  delay: 5
  when: ksql_ssl_mutual_auth_enabled|bool
  ignore_errors: true

- name: Fetch Files for Debugging Failure
  include: tasks/failure_handling.yml
  vars:
    service_name: "{{ksql_service_name}}"
    config_file: "{{ksql.config_file}}"
  # Only one health check runs, only one will have a 'failed' field. For skipped checks, defaulting 'failed' to False
  when: result1.failed|default(False) or result2.failed|default(False)
